packages/ag-charts-enterprise/src/axes/angle-number/angleNumberAxis.ts

```diff
@@ -0,0 +1,468 @@
+import type { AgAngleAxisLabelOrientation, _Scale } from 'ag-charts-community';
+import { _ModuleSupport, _Scene, _Util } from 'ag-charts-community';
+import { AngleCrossLine } from './angleCrossLine';
+import { PolarLinearScale } from '../polarLinearScale';
+
+const {
+    AND,
+    assignJsonApplyConstructedArray,
+    ChartAxisDirection,
+    Default,
+    GREATER_THAN,
+    LESS_THAN,
+    NUMBER,
+    NUMBER_OR_NAN,
+    predicateWithMessage,
+    ProxyOnWrite,
+    Validate,
+} = _ModuleSupport;
+const { Path, Text } = _Scene;
+const { isNumberEqual, toRadians, normalizeAngle360, normalisedExtentWithMetadata } = _Util;
+
+interface AngleNumberAxisLabelDatum {
+    text: string;
+    x: number;
+    y: number;
+    hidden: boolean;
+    rotation: number;
+    textAlign: CanvasTextAlign;
+    textBaseline: CanvasTextBaseline;
+    box: _Scene.BBox | undefined;
+}
+
+interface AngleNumberAxisTickDatum {
+    value: number;
+    visible: boolean;
+}
+
+function loopSymmetrically<T>(items: T[], step: number, iterator: (prev: T, next: T) => any) {
+    const loop = (start: number, end: number, step: number, iterator: (prev: T, next: T) => any) => {
+        let prev = items[0];
+        for (let i = start; step > 0 ? i <= end : i > end; i += step) {
+            const curr = items[i];
+            if (iterator(prev, curr)) return true;
+            prev = curr;
+        }
+    };
+
+    const midIndex = Math.floor(items.length / 2);
+
+    if (loop(step, midIndex, step, iterator)) return true;
+    return loop(items.length - step, midIndex, -step, iterator);
+}
+
+const ANGLE_LABEL_ORIENTATIONS: AgAngleAxisLabelOrientation[] = ['fixed', 'parallel', 'perpendicular'];
+
+const ANGLE_LABEL_ORIENTATION = predicateWithMessage(
+    (v: any) => ANGLE_LABEL_ORIENTATIONS.includes(v),
+    `expecting a label orientation keyword such as 'fixed', 'parallel' or 'perpendicular'`
+);
+
+class AngleCategoryAxisLabel extends _ModuleSupport.AxisLabel {
+    @Validate(ANGLE_LABEL_ORIENTATION)
+    orientation: AgAngleAxisLabelOrientation = 'fixed';
+}
+
+export class AngleNumberAxis extends _ModuleSupport.PolarAxis<_Scale.LinearScale> {
```

Comment:
This entire class looks very similar to `angleCategoryAxis.ts` - can we factor out the common behaviours?